src/components/marketing-table.ts

```diff
@@ -0,0 +1,80 @@
+import {Paginator} from "../controller/paginator";
+import {PaginatorModel} from "../models/pagination/pagination-model";
+import {DefaultPaginationView} from "../views/pagination/pagination-view";
+import {get} from "../helpers/fetch";
+import {Listener} from "../event-manager/interfaces";
+import { Filter } from "../controller/filter";
+import { FilterModel } from "../models/filter/filter-model";
+import { DefaultFilterView } from "../views/filter/filter-view";
+import { FilterConfigItem, FilterConfig } from '../models/filter/filter-config-item';
+import { Table } from "../controller/table";
+import { TableModel } from "../models/table/table-model";
+import { DefaultTableView } from "../views/table/table-view";
+import { RowModelItem, RowConfig } from "../models/table/row-model-item";
+import {MarketingItem} from "../data-models/marketing-item";
+
+export class MarketingTable implements Listener {
+    paginator: Paginator<MarketingItem>;
+    filter: Filter<MarketingItem>;
+    table: Table<MarketingItem>;
+    rowConfig: RowConfig<MarketingItem>;
+
+    constructor() {
+        this.paginator = new Paginator<MarketingItem>(new PaginatorModel(), new DefaultPaginationView('.marketing-table .pagination'));
+        this.paginator.attach('pagiChange', this);
+        this.paginator.initNewData(get('marketing'));
+
+        let filterConfig: FilterConfig = {
+            displays: new FilterConfigItem('Displays'),
+            clicks: new FilterConfigItem('Clicks '),
+            orders: new FilterConfigItem('Purchases'),
+            addToCarts: new FilterConfigItem('Add to cart'),
+            units: new FilterConfigItem('Sold units'),
+            revenue: new FilterConfigItem('Revenue'),
+            profit: new FilterConfigItem('Profit')
+        };
+
+        this.filter = new Filter<MarketingItem>(new FilterModel(), new DefaultFilterView('.marketing-table .filter'));
+        this.filter.attach('filterChange', this);
+        this.filter.initNewData(get('marketing'), filterConfig);
+
+        this.rowConfig = {
```

Comment:
The same can be done here:
```
It would be nice to keep this in a separate file with constants or above the class.
```typescript
MARKETING_REPORT_COLUMNS: ColumnsConfig = { // ... };
```
```